### ISSUE TITLE
"Failed to generate flow"

### ISSUE THREAD

**Issue by cobolcowboy** [BroadcomMFD/cobol-control-flow#2]
Every file I've tried this on fails and I just get a little notification popup that says "Failed to generate flow". I can't find any other information on why its failing. I'm on Win10 x64 and the latest version of VSCode. This extension looks like it could be very helpful if it worked. Is there any way I could produce some error output or anything?

**Comment by cobolcowboy** [comment]
I'm using the Bitlang COBOL syntax extension...do I need to use yours to utilize this extension?

**Comment by venkatzhub** [comment]
@cobolcowboy - Yes, we recommenced you install the COBOL Language support: https://marketplace.visualstudio.com/items?itemName=broadcomMFD.cobol-language-support

**Comment by cobolcowboy** [comment]
I tried installing that but the extension still failed to generate flow. I don't see any errors in any of the logs....is there any way to get more information about why it is failing?

**Comment by vibhor-bc** [comment]
@cobolcowboy Can you try generating flow for the sample files ?
[https://github.com/BroadcomMFD/cobol-control-flow/tree/master/sample](url)

Also the logs can be accessed via the developer tools -
1). Open the COBOL file for analysis.
2). Ctrl + Shift + P
3). Type "Developer:Toggle Developer Tools"
4).  Go to console tab and clear any existing logs.
5). Hit "View COBOL Contorl Flow".
6). If the analysis fails , you should be able to see the logs in red.

**Comment by cobolcowboy** [comment]
Thank you, I see now it is an issue with the format of my files. Most of our source is not tabbed correctly according to COBOL standards. Usually we just start at column 0. I tried using the Format Document option and that doesn't seem to add the whitespace. I suppose I could add it manually to generate the flow.

**Comment by vibhor-bc** [comment]
Closing this issue as the problem was due to the format of the file(tabbing of source).

**Comment by venkatzhub** [comment]
@cobolcowboy - we are adhering to and supporting the z/OS COBOL syntax and honoring the Area A, Area B. Do reach out if you continue to have issues with the source indented properly.

**Comment by cobolcowboy** [comment]
I don't want to open a new issue for this but I was hoping one of you might be kind enough to provide your input. I'm trying to get this extension to support the free format. I don't use Java often or ANTLR ever. I tried dropping in grammars that I know work with the free format but the code appears to rely on specific references from the ANTLR generated classes (again, sry, noob with this stuff) so I can't just drop in a replacement. I asked in the ##antlr irc though and they said that the grammars (the ones from this extension) aren't enforcing the whitespace at the start of the line. So is it in the implementation somewhere?

Without the whitespace at the start of the line I get an invalid syntax error which looks like it only is raised from the ANTLR generated classes which lead me to believe that was encoded into the grammar somewhere. I did also notice the regex and if statements regarding line length in the line reader class but it didn't look like that would actually cause an error but I'm not entirely sure.

Anyways thanks very much for any input.
